src/main/java/taxi/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,28 @@
+package taxi.service.impl;
+
+import java.util.Optional;
+import taxi.exception.AuthenticationException;
+import taxi.lib.Inject;
+import taxi.lib.Service;
+import taxi.model.Driver;
+import taxi.service.AuthenticationService;
+import taxi.service.DriverService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverService driverService;
+
+    @Override
+    public Driver login(String drivername, String password)
+            throws AuthenticationException {
+        Optional<Driver> foundByloginDriver = driverService.findByLogin(drivername);
```

Comment:
I think this variable should be of type `Driver`
```suggestion
        Driver foundByloginDriver = driverService.findByLogin(drivername);
```
But I am not sure